### PR TITLE
Update use use loggo/v2 and use log labels as a map of strings

### PR DESCRIPTION
**Pull request by wallyworld**
This PR brings in upstream loggo v2 which changes the log Entry struct to use a string map of labels.
To make it all work, 5 upstream dependencies needed to be changed, causing a lot of import churn. These are landed in separate commits:
- [loggo/v2](https://github.com/juju/juju/pull/16908/commits/568c6815c8531b9b3f8383f4048c15f4b6194ede)
- [charm/v13](https://github.com/juju/juju/pull/16908/commits/be7166ac09dce0154280c66c99f80e203d3de217)
- [utils/v4, testing v1.2.0](https://github.com/juju/juju/pull/16908/commits/3ff63137178d75b0ef96ee08a4f0691f734c11cb)
- [cmd/v4](https://github.com/juju/juju/pull/16908/commits/67d89ab27eb0af6d5dcc53b26345fd32e495f10b)

The real work for this PR is in the [last commit](https://github.com/juju/juju/pull/16908/commits/db8b803fe09bdf009b71a7d60da3f9d0cf4082ea). The `logger.ChildWithLables` call is renamed to `logger.ChildWithTags`.  The log record Labels map is used instead of array of strings. 

Some legacy log record version attributes are removed since these are only used by the log forward worker (which is being deleted). There's even an ancient expired bug about this: https://bugs.launchpad.net/juju/+bug/1643743.

A future PR will need to deal with over the wire compatibility for the 2 labels formats.

Because logs in state is going away, some state tests using the old labels were deleted rather than wasting time porting them to use a map of strings.

## QA steps

`juju bootstrap --config logging-config="#charmhub=DEBUG" ...`

Logger tags are now in the debug log output as `logger-tags:tag1,tag2`

```
machine-0: 10:11:00 DEBUG juju.apiserver.charmdownloader.charmdownloader logger-tags:charmhub downloading charm "ubuntu" from requested origin {charm-hub charm   0xc000eddda8 stable amd64/ubuntu/22.04 }
```

Note - for now, the log lines are still space separated values. Subsequent PRs will change this to newline delimited json.

Migrate a model, check logs after migration.

## Links

**Jira card:** JUJU-5434

### CODE REVIEW

File: rpc/params/logstream.go [hunk 1]
```diff
@@ -7,24 +7,27 @@ import (
...
+	ID           int64             `json:"id"`
+	ModelUUID    string            `json:"mid"`
+	Entity       string            `json:"ent"`
+	Timestamp    time.Time         `json:"ts"`
+	Module       string            `json:"mod"`
+	Location     string            `json:"lo"`
+	Level        string            `json:"lv"`
+	Message      string            `json:"msg"`
+	LegacyLabels []string          `json:"lab,omitempty"`
+	Labels       map[string]string `json:"lbl,omitempty"`
```
Comment (hpidcock) [inline rpc/params/logstream.go:30]: that's one way

File: rpc/params/params.go [hunk 2]
```diff
@@ -1020,14 +1020,17 @@ type RebootActionResult struct {
...
-	Message  string    `json:"x"`
-	Entity   string    `json:"e,omitempty"`
-	Labels   []string  `json:"c,omitempty"`
+	Time     time.Time         `json:"t"`
+	Module   string            `json:"m"`
+	Location string            `json:"l"`
+	Level    string            `json:"v"`
+	Message  string            `json:"x"`
+	Entity   string            `json:"e,omitempty"`
+	Labels   map[string]string `json:"b,omitempty"`
```
Comment (hpidcock) [inline rpc/params/params.go:1033]: hope we don't run out of letters anytime soon.

File: testing/cert.go [hunk 3]
```diff
@@ -63,7 +63,22 @@ func mustParseServerCert(srvCert string, srvKey string) *tls.Certificate {
 }
 
 func mustParseCertAndKey(certPEM, keyPEM string) (*x509.Certificate, *rsa.PrivateKey) {
-	cert, key, err := utilscert.ParseCertAndKey(certPEM, keyPEM)
+	tlsCert, err := tls.X509KeyPair([]byte(certPEM), []byte(keyPEM))
```
Comment (hpidcock) [inline testing/cert.go:66]: Nice. Using standard library features directly... how dare you 😆

**Review by hpidcock — APPROVED** [review]

File: rpc/params/logstream.go (hunk 1 above)
Comment (wallyworld) [reply rpc/params/logstream.go:30]: ah, meant to not use LegacyLabels for now. will delete that. another PR will do the compatibility stuff

File: testing/cert.go (hunk 3 above)
Comment (wallyworld) [reply testing/cert.go:66]: this is because our utils lib now does ed25519 and mongo still uses rsa :-)

**Pull request merged**